### ISSUE TITLE
Is scrolling supposed to be non-fucntional in the layers pane? (running through wine)

### ISSUE THREAD

**Issue by drazil100** [ChrisMiuchiz/Plasma-Writeup#2]
I am currently without a windows partition so I have no way of testing whether this is a bug / unimplemented feature with plasma or if its a bug with wine but I can't scroll through the layers pane.

On a side note do you have even the foggiest idea how layers are _supposed_ to work? I feel like im playing a game of whack-a-mole with my layers. I lock or unlock one layer and several others lock or unlock and sometimes moving one layer up brings a completely different layer to the top etc... If there is a specific way I should be adding objects to a sheet that would make layers behave in a remotely predictable way id love to know how to do things right.

Thanks a lot for all your hard work reverse engineering this stuff by the way. I have wanted to try plasma since cube world first hit alpha but sadly I missed the initial preview. I am glad I finally get the chance to try it out! :P

**Comment by coremaze** [comment]
I can scroll through the layers pane fine on Windows and Linux.

I haven't had layers lock or unlock each other except for cases where a layer is a child of another layer. Double clicking a layer also sets that one as a parent, so any new meshes you draw will be a child of that one.

**Comment by drazil100** [comment]
Any idea what I can do to get scrolling to work in wine?

**Comment by coremaze** [comment]
What about scrolling doesn't work? Does dragging the scrollbar work? Does the scroll wheel work? If not, does zooming on sheets using the scroll wheel work?

**Comment by drazil100** [comment]
Not only does the scroll wheel not work the scroll bars handles are max size all the time and can’t be dragged at all. Items just go off the bottom of the pane and literally the only way to access them is to resize the pane itself. I’m experiencing this on both Linux mint 19.3 and manjaro by the way.

**Comment by coremaze** [comment]
Since I don't have this issue on any configuration it might be related to a hardware difference, but the only thing I could see impacting this is resolution. This software works fine for me and I normally run it maximized on 1920x1080 displays.

**Comment by drazil100** [comment]
Both of my displays are 1366x768. Also I am running plasma on 2 separate machines. If there is any additional information I can help provide let me know.

Also here is a screenshot of it misbehaving.
![Screenshot from 2020-07-06 16-25-41](https://user-images.githubusercontent.com/17794391/86647455-afe8cc80-bfa5-11ea-978a-7927d1ba0c83.png)

**Comment by coremaze** [comment]
I'm not sure how else to try to debug this problem without having to go too far into RE, however I have shown that it works at the similar resolution of 1368x768 on a debian-based OS through wine.

![image](https://user-images.githubusercontent.com/1361206/86649447-c4ca5d80-bfaf-11ea-987f-677884d95fc7.png)

This probably requires additional reverse engineering to diagnose, but I can't replicate the issue.

**Comment by drazil100** [comment]
What distro are you running it on? Also can you tell me what version of wine you are running? Also exactly what do you have installed to run it? 

Maybe if you can provide a setup that is known to work I can figure out how to replicate the setup on the distros I use.

**Comment by coremaze** [comment]
Elementary OS 5.1 Hera (Built on Ubuntu 18.04 LTS)

wine-5.0 (though I believe it also worked on older versions)

I only needed to bring my own Windows fonts for text to render, and to apply the PlasmaTweaks update for [pretty much anything to render properly](https://www.youtube.com/watch?v=FBkj3ZUs0xw)

**Comment by coremaze** [comment]
I also noticed that your screenshot does not have layer names being displayed at all

**Comment by coremaze** [comment]
I've replicated this bug by removing the tahoma font.

![image](https://user-images.githubusercontent.com/1361206/86843629-d32e7d00-c074-11ea-9d87-27a001349d2d.png)

Clearly you have _some_ of the required fonts, so I don't know why this one is missing, but you should acquire a full set of Windows fonts before trying to use GUI applications on wine.

**Comment by drazil100** [comment]
I was advised to install corefonts with winetricks especially considering I don’t have a windows partition. Guessing it doesn’t have every font though. I’ll look into it though. Thanks.

**Comment by drazil100** [comment]
Do you know if missing literally any font will break it or are there only specific fonts that cause breakage? If only certain fonts are required would it be possible to maybe get a list of exactly which fonts are required?

**Comment by coremaze** [comment]
Looking for TTFs in the strings of the binary, it seems to use at least the following, with the names in the Layers pane using one of the tahomas (I didn't test which one)

verdanab.ttf
arial.ttf
verdana.ttf
tahomabd.ttf
tahoma.ttf

**Comment by drazil100** [comment]
Realistically I THINK I should have all those. I know I have at least tahoma installed. Used tahoma for my streaming layout so I archived that one personally and is on at least one of the machines. Not sure if it needs to somehow be installed within wine or if it can be installed on the linux side (or if its the same thing)...

Regardless I will double check that I have all of those somewhere on my system and get back to you with my findings.

**Comment by coremaze** [comment]
The location I removed the tahoma fonts from in order to replicate the bug was ~/.wine/drive_c/windows/Fonts. If you end up with a Windows installation sometime, you should copy all the fonts to there from C:\Windows\Fonts.

**Comment by drazil100** [comment]
Wow I got SUPER lucky that the one windows font NOT in corefonts that I needed to get plasma to work right was the one windows font I actually manually backed up on google drive for my streaming layout... Put them in ~/.wine/drive_c/windows/Fonts and it works now. Thanks! 

Now if only we can remove the need for an authentication server... Thanks again for all your hard work! Plasma is fantastic and I am glad someone cared enough to do what you did.
